consensus/istanbul/backend/backend.go

```diff
@@ -794,3 +818,44 @@ func (sb *Backend) ConnectToVals() {
 		sb.RefreshValPeers(valset)
 	}
 }
+
+func (sb *Backend) retrieveRegisteredAndElectedValidators() (map[common.Address]bool, error) {
```

Comment:
I knew you were going to mention this :).

I was a bit worried about inconsistent reads.  e.g. if the reg/elected valset is cached from a few blocks ago, what's the impact of that to user of that cache.  I'll do some thinking about this.

Reply:
The node is now caching the reg/elected valset such that it's retrieved at most once a minute.  Here's the commit:  https://github.com/celo-org/celo-blockchain/pull/816/files/5f32145a33146ce1b79c123a4a542146596cce02..ab4fdb2526f16a22611261847372a885088fbafb

I also made changes so that the first gossiped announce message will be delayed by 1 minute.  This will be so that all of the receivers of that message will not use a stale valset (which may cause them to not process the gossiped message).

Reply:
Ok, thanks for making this change!